tests/staking/Migrator.t.sol

```diff
@@ -0,0 +1,413 @@
+        }
+
+        vm.stopPrank();
+    }
+
+    function testMigrateStakesWithOGNBalance() public {
+        vm.startPrank(alice);
+        ogvStaking.mockStake(10000 ether, 365 days);
+        ogvStaking.mockStake(1000 ether, 20 days);
+
+        ogn.mint(alice, 500 ether);
+
+        uint256 ognBalanceBefore = ogn.balanceOf(alice);
+
+        uint256[] memory lockupIds = new uint256[](2);
+        lockupIds[0] = 0;
+        lockupIds[1] = 1;
+
+        uint256 stakeAmount = ognBalanceBefore + ((11000 ether * 0.09137 ether) / 1 ether);
+
+        migrator.migrate(lockupIds, 0, 500 ether, false, stakeAmount, 300 days);
+
+        // Should have merged it in a single OGN lockup
+        (uint128 amount, uint128 end, uint256 points) = ognStaking.lockups(alice, 0);
+        assertEq(amount, stakeAmount, "Lockup not migrated");
+
+        // Should have updated balance correctly
+        assertEq(ogn.balanceOf(alice), 0, "OGN Balance mismatch");
+
+        vm.expectRevert();
+        (amount, end, points) = ognStaking.lockups(alice, 1);
+
+        // Should have removed OGV staked
+        for (uint256 i = 0; i < lockupIds.length; ++i) {
+            (amount, end, points) = ogvStaking.lockups(alice, lockupIds[i]);
+            assertEq(amount, 0, "Lockup still exists");
+            assertEq(end, 0, "Lockup still exists");
+            assertEq(points, 0, "Lockup still exists");
+        }
+
+        vm.stopPrank();
+    }
+
+    function testMigrateStakesWithOGNAndOGVBalances() public {
+        vm.startPrank(alice);
+        ogvStaking.mockStake(10000 ether, 365 days);
+        ogvStaking.mockStake(1000 ether, 20 days);
+
+        ogn.mint(alice, 500 ether);
+
+        uint256 ognBalanceBefore = ogn.balanceOf(alice);
+        uint256 ogvBalanceBefore = ogv.balanceOf(alice);
+
+        uint256[] memory lockupIds = new uint256[](2);
+        lockupIds[0] = 0;
+        lockupIds[1] = 1;
+
+        uint256 stakeAmount = ognBalanceBefore + ((11500 ether * 0.09137 ether) / 1 ether);
+
+        migrator.migrate(lockupIds, 500 ether, ognBalanceBefore, false, stakeAmount, 300 days);
+
+        // Should have merged it in a single OGN lockup
+        (uint128 amount, uint128 end, uint256 points) = ognStaking.lockups(alice, 0);
+        assertEq(amount, stakeAmount, "Lockup not migrated");
+
+        // Should have updated balance correctly
+        assertEq(ogn.balanceOf(alice), 0, "OGN Balance mismatch");
+        assertEq(ogv.balanceOf(alice), ogvBalanceBefore - 500 ether, "OGN Balance mismatch");
+
+        vm.expectRevert();
+        (amount, end, points) = ognStaking.lockups(alice, 1);
+
+        // Should have removed OGV staked
+        for (uint256 i = 0; i < lockupIds.length; ++i) {
+            (amount, end, points) = ogvStaking.lockups(alice, lockupIds[i]);
+            assertEq(amount, 0, "Lockup still exists");
+            assertEq(end, 0, "Lockup still exists");
+            assertEq(points, 0, "Lockup still exists");
```

Comment:
good to add a different string as it will be easier to figure out which one fails, if one if them fail.

```solidity
  assertEq(amount, 0, "amount:Lockup still exists");
  assertEq(end, 0, "end:Lockup still exists");
  assertEq(points, 0, "points:Lockup still exists");
```